Fix hierarchical logging documentation

AhmedLSayed9 opened the pull request:
The hierarchical logging documentation is inaccurate and needs to be updated.
_Note_: Logging level should enable logging for all levels above that level not below that level.

- Thanks for your contribution! Please replace this text with a description of what this PR is changing or adding and why, list any relevant issues, and review the contribution guidelines below.

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/blob/main/docs/External-Package-Maintenance.md#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

devoncarew commented:
Thanks for the contribution. I think the existing example is too long and seems to be trying to demonstrate a few different things. I think is should be shortened quite a bit in order to just show very simple API usage. This might mean in-lining the sample from the example/ dir, or perhaps just eliding the code sample in the `Configuration` section completely.

The existing example does look correct however - both the comments in the code and the output as quoted.

AhmedLSayed9 commented:
> Thanks for the contribution. I think the existing example is too long and seems to be trying to demonstrate a few different things. I think is should be shortened quite a bit in order to just show very simple API usage. This might mean in-lining the sample from the example/ dir, or perhaps just eliding the code sample in the `Configuration` section completely.
> 
> The existing example does look correct however - both the comments in the code and the output as quoted.

I don't think it's all correct. I've simplified and updated the changes to clarify more about the incorrect behavior/output.
I'll explain them one by one in the following comments.

AhmedLSayed9 commented on README.md:
```diff
@@ -108,34 +108,30 @@ their `onRecord` streams.
 
-  // Will NOT print because FINER is too low level for `Logger.root`.
+  // Will NOT print because FINER is too low level for ([LOG1] & [ROOT]).
   log1.finer('LOG_01 FINER (X)');
```
This one is correct, I've just updated the comment to be more clear.

AhmedLSayed9 commented on README.md:
```diff
@@ -108,34 +108,30 @@ their `onRecord` streams.
   log1.finer('LOG_01 FINER (X)');
 
-  // Will print twice ([LOG1] & [ROOT])
-  log1.fine('LOG_01 FINE (√√)');
```
This one is incorrect.
It should print once by [LOG1] only as the root logger has higher level `WARNING`.
As per [Level](https://pub.dev/documentation/logging/latest/logging/Level-class.html) definiton, logging level should enable logging only for all levels above the level not below it.

However, the output shows that the root logger fires which is actually a bug https://github.com/dart-lang/logging/issues/145

AhmedLSayed9 commented on README.md:
```diff
@@ -108,34 +108,30 @@ their `onRecord` streams.
+  log1.fine('LOG_01 FINE (√)');
 
-  // Will print ONCE because `log1` only uses root listener.
-  log1.warning('LOG_01 WARNING (√)');
```
This one is incorrect.
It should print twice as `WARNING` is sufficient for both [LOG1] & [ROOT], and the output actually print it twice as:
```
[LOG1][FINE+] LOG_01 WARNING (√)
[ROOT][WARNING+] LOG_01 WARNING (√)
```

AhmedLSayed9 commented on README.md:
```diff
@@ -108,34 +108,30 @@ their `onRecord` streams.
 
-  // Will never print because FINE is too low level.
+  // Will NOT print because FINE is too low level for ([LOG1] & [ROOT]).
   log2.fine('LOG_02 FINE (X)');
```
This one is correct, I've just updated the comment to be more clear.

AhmedLSayed9 commented on README.md:
```diff
@@ -108,34 +108,30 @@ their `onRecord` streams.
 
   // Will print twice ([LOG2] & [ROOT]) because warning is sufficient for all
   // loggers' levels.
   log2.warning('LOG_02 WARNING (√√)');
```
This one is correct.

AhmedLSayed9 commented on README.md:
```diff
@@ -108,34 +108,30 @@ their `onRecord` streams.
-
-  // Will never print because `info` is filtered by `Logger.root.level` of
-  // `Level.WARNING`.
-  log2.info('INFO (X)');
```
This one is correct but it's not needed as it's the same case as `log2.fine('LOG_02 FINE (X)');`

mosuem commented:
Closing as the [dart-lang/logging](https://github.com/dart-lang/logging) repository is merged into the [dart-lang/core](https://github.com/dart-lang/core) monorepo. Please re-open this PR there!
